File: projecteuler/0 - 100/003/largestPrimeFactor.py
from math import sqrt, ceil

# Check if a number is prime
def isPrime(n):
    for i in range(2, ceil(sqrt(n))):
        if(int(n) % i == 0):
            return False
    return True
# Find the largest prime number below n
def largestPrime(n):
    largestPrime = 1
    for i in range(1, int(n)):
        if(isPrime(i) and i > largestPrime and n % i == 0):
            largestPrime = i
    return largestPrime

if __name__ == "__main__":
    # Get user input
    n = input("Enter an integer n: ")
    # Print results
    print("\nLargest Prime below {:,}".format(int(n)))
    print("{:,}".format(largestPrime(n)))

File: projecteuler/0 - 100/003/largestPrimeFactor.scala
object largestPrimeFactor {
    // Determine the factors of int n
    def factors(n: Int): List[Int] = {
        var factors = List(1)
        for(i <- 2 to n) {
            if(n % i == 0) { factors = factors ::: List(i) }
        }
        return factors
    }
    // Determine if int n is prime
    def isPrime(n: Int): Boolean = {
        for(i <- 2 to n - 1) {
            if(n % i == 0) { return false }
        }
        return true
    }
    def main(args: Array[String]) {
        val n = args(0)
        println(s"\nLargest Prime below $n:")
        var largestPrime = 1
        val f = factors(n.toInt)
        println(s"Factors: ${f}")
        f.foreach(i => {
            if(isPrime(i) && i > largestPrime) { largestPrime = i }
        })
        println(s"Largest Prime Factor: ${largestPrime}")
    }
}